### DISCUSSION TITLE
Using Clip + Flip seems to use the data outside of the clip region, how do I avoid that?

### DISCUSSION THREAD

**Discussion by Thraka** [SixLabors/ImageSharp#2573] [Q&A]
I'm trying to flip some sprites contained within a sprite sheet, but it doesn't seem to work as expected. I'm doing these things:

1. get the path rectangle of where my sprite is located
2. clip the context to that rectangle
3. apply a flip to the clip context

It seems that flip is restricted to the clip for the output of the flip operation, but it's using the entire image as the input. How do I restrict flip to use the clip region as the source and target of the operation?

```csharp
texture.Mutate(context =>
{
    var options = context.Configuration.GetGraphicsOptions();
    options.Antialias = false;
    context.Configuration.SetGraphicsOptions(options);


    // Rectangle from other library I'm using
    var fontGlyphRect = font.GetGlyphSourceRectangle(1);

    // Transform into SixLabors rectangle
    RectangleF rect = new(fontGlyphRect.X, fontGlyphRect.Y, fontGlyphRect.Width, fontGlyphRect.Height);

    // Convert rect to IPath to clip, then flip that clipped area
    context.Clip(new RectangularPolygon(rect), context2 => context2.Flip(FlipMode.Vertical));
});
```

**Comment by JimBobSquarePants** [comment]
`Clip()` is part of ImageSharp.Drawing which uses drawing specific `IPath` types and brushes so it cannot be used in this manner. What we need to do is to add an overload to `Flip()` that takes a `Rectangle` bounds.
